File: pset6/readability.py
from cs50 import get_string

# Main function


def main():
    # Get text from user
    text = get_string("Text: ")

    # Count letters, words, sentences
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # count L and S
    L = 100.0 * (letters/words)
    S = 100.0 * (sentences/words)

    # Calculate index
    index = 0.0588 * L - 0.296 * S - 15.8

    # Print grade
    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade {}".format(round(index)))


# Counts letters
def count_letters(text):
    letter_counter = 0

    for x in text:
        c = x.upper()
        if c >= 'A' and c <= 'Z':
            letter_counter += 1

    print(letter_counter)
    return letter_counter


# Counts words
def count_words(text):
    word_counter = 1

    for x in text:
        if x == ' ':
            word_counter += 1

    print(word_counter)
    return word_counter


# Counts sentences
def count_sentences(text):
    sentence_counter = 0

    for x in text:
        if x == '.':
            sentence_counter += 1
        elif x == '!':
            sentence_counter += 1
        elif x == '?':
            sentence_counter += 1

    print(sentence_counter)
    return sentence_counter


# Calls main function if main function exist
if __name__ == "__main__":
    main()

File: pset2/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

//Prototypes
int count_letter(string str);
int count_words(string str);
int count_sentences(string str);



int main(void)
{
    //Get input
    string s = get_string("Text: ");


    //Count
    int letters = count_letter(s);

    int words = count_words(s);

    int sentences = count_sentences(s);

    //Math
    double L = 100 * ((double)letters / (double)words);

    double S = 100 * ((double)sentences / (double)words);

    //Coleman-Liau index Equation
    double index = 0.0588 * L - 0.296 * S - 15.8;

    int rounded_index = round(index);




    //Print grade
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", rounded_index);
    }




}



//Count total letters of the text
int count_letter(string str)
{
    int sum = 0;

    for (int i = 0 ; str[i] != 0; i++)
    {
        if (str[i] >= 'a' && str[i] <= 'z')
        {
            sum = sum + 1;
        }
        else if (str[i] >= 'A' && str[i] <= 'Z')
        {
            sum = sum + 1;
        }

    }

    return sum;
}


//count number of words
int count_words(string str)
{
    int total = 0;

    for (int i = 0 ; str[i] != 0; i++)
    {
        if (str[i] == 32)
        {
            total = total + 1;
        }
    }

    return total + 1;
}



//Count sentences
int count_sentences(string str)
{
    int summ = 0;

    for (int i = 0 ; str[i] != 0; i++)
    {
        if (str[i] == 46)
        {
            summ = summ + 1;
        }
        else if (str[i] == 33)
        {
            summ = summ + 1;
        }
        else if (str[i] == 63)
        {
            summ = summ + 1;
        }


    }

    return summ;
}